### ISSUE TITLE
django-hosts subdomains not detected

### ISSUE THREAD

**Issue by pitchopp** [tfranzel/drf-spectacular#1334]
Hello, i installed drf_spectacular in my project that uses django-hosts for handling multiple subdomains. spectacular doesn't detect any view because my ROOT_URLCONF is empty.

here is my django_hosts configuration

```python
from django_hosts import patterns, host
from django.conf import settings

host_patterns = patterns('',
    host(r'', settings.ROOT_URLCONF, name=' '),
    host(r'api', 'api.urls', name='api'),
    host(r'admin', 'backend.admin_urls', name='admin')
)
```

my root urls is empty because my routes are located in "api" app

```python
urlpatterns = []
```

here is the output of ./manage.py spectacular

```bash
> python manage.py spectacular

openapi: 3.0.3
info:
  title: Your Project API
  version: 1.0.0
  description: Your project description
paths: {}
components: {}
```

Am i missing a configuration to tell spectacular how to fetch other hosts urls ?

**Comment by tfranzel** [comment]
Hi, I was unaware of this library. Seems like it completely bypasses the regular `urlconf` and instead injects its own routes via middleware. Spectacular&DRF both know nothing about this "hack". 

I guess you have to make spectacular aware of thos injected routes by customizing the generator:

https://github.com/tfranzel/drf-spectacular/blob/27ae6f293a6e8731649216d62ba2a5c8d65691e6/drf_spectacular/settings.py#L30

In fact you only want to set a customized `Enumerator` class and leave the rest as is:

https://github.com/tfranzel/drf-spectacular/blob/27ae6f293a6e8731649216d62ba2a5c8d65691e6/drf_spectacular/generators.py#L103

You then subclass our enumerator and overwrite the `__init__` and make it aware of those other routes.

https://github.com/encode/django-rest-framework/blob/dbac145638758413b966c3418fa5f3f651e3e02a/rest_framework/schemas/generators.py#L61

So basically one changed setting, 1 class with one changed attribute and 1 class with a patched __init__.
